### ISSUE TITLE
Support for gatsby 4?

### ISSUE THREAD

**Issue by simplenotezy** [deckgo/gatsby-remark-highlight-code#55]
I managed to install highlight code without issues; but after upgrading npm I now get this error:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: constant-solutions@1.0.0
npm ERR! Found: gatsby@4.4.0
npm ERR! node_modules/gatsby
npm ERR!   gatsby@"^4.4.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer gatsby@"^3.0.0" from gatsby-remark-highlight-code@3.1.0
npm ERR! node_modules/gatsby-remark-highlight-code
npm ERR!   gatsby-remark-highlight-code@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
```

**Comment by peterpeterparker** [comment]
Have you try cleaning npm (`rm package-lock.json && rm -r node_modules && npm i`)?

I can give it a try if you provide a sample repo to reproduce the error but the issue does not seem to be related to this plugin code but rather something on your project end.

**Comment by Veuge** [comment]
Same here, I'm having the warning
```
warn Plugin gatsby-remark-highlight-code is not compatible with your gatsby version 4.4.0 - It requires gatsby@^3.0.0
```

on `package.json` I have
```
"gatsby-remark-highlight-code": "^3.1.0",
```

**Comment by peterpeterparker** [comment]
1. Is there any issue or is that "just" a warning?
2. Should we "just" bump the minimal requirement from gatsby v3 to v4 or is there any breaking changes affecting the plugin?
3. Can you provide a sample repo to reproduce the issue?

**Comment by DylanLacey** [comment]
**1: **
This generates an `ERR!` not just a `WARN`:

```
npm ERR! While resolving: highlight-sad@1.0.0
npm ERR! Found: gatsby@4.4.0
npm ERR! node_modules/gatsby
npm ERR!   gatsby@"^4.4.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer gatsby@"^3.0.0" from gatsby-remark-highlight-code@3.1.0
npm ERR! node_modules/gatsby-remark-highlight-code
npm ERR!   gatsby-remark-highlight-code@"^3.1.0" from the root project
```

**2:**
I was able to get my site up and running by adding an override to my package.json:

```
  "overrides": {
    "gatsby-remark-highlight-code": {
      "gatsby": "4.4.0"
    }
  }
```

That implies to me that there aren't any breaking changes, but I'm no Gatsby expert.


For anyone reading this bug: **This is a terrible idea.  Don't do this.  Wait until it's supported.**

**3:**
Please see https://github.com/DylanLacey/gatsby-remark-code-4 for a minimal repro; Just run `npm install`.

**Comment by peterpeterparker** [comment]
Migration guide from Gatbsy v3 to v4:

- [Updating community plugins](https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v3-to-v4/#updating-community-plugins)

- [For Plugin Maintainers](https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v3-to-v4/#for-plugin-maintainers)

(notes to my self, on it)

**Comment by peterpeterparker** [comment]
Awesome, thanks a lot @DylanLacey 🙏

Did a quick try with your repo (including writing some code in a .mdx file), the plugin is also compatible with Gatsby v4.
Therefore I just added the new `peerDependencies` in addition to the existing support of v3 in `package.json`.

This has been released to npm in version `v3.2.0`.

```
npm rm gatsby-remark-highlight-code && npm i gatsby-remark-highlight-code
```

If anyone notice issue with Gatsby v4, please let me know.

**Comment by simplenotezy** [comment]
Thanks a lot @peterpeterparker  - great work!
